goreleaser/docker-build-pipelines.yml

```diff
@@ -0,0 +1,50 @@
+dist: /tmp/pachyderm/release/docker-build-pipeline
+
+before:
+  hooks:
+    - docker login
+
+builds:
+  -
+    id: pachctl
+    dir: src/server/cmd/pachctl
+    main: main.go
+    binary: pachctl
+    ldflags:
+      - -X {{ .Env.LD_FLAGS }} -X main.commit={{.Commit}} -X main.date={{.Date}} -X main.builtBy=goreleaser
+    gcflags:
+      - all=-trimpath={{ .Env.PWD }}
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+    goarch:
+      - amd64
+
+archives:
+  - format: binary
+
+checksum:
+  disable: true
+
+changelog:
+  skip: true
+
+release:
+  disable: true
+
+dockers:
+  -
+    image_templates:
+      - pachyderm/go-build:{{.Env.VERSION}}
```

Comment:
Why {{.Env.VERSION}} throughout and not {{.Version}}?  I worry that there are two sources for which version is being built; goreleaser's own detection based on tags, and the environment.  It would be most reliable to consistently populate one from the other.  (i.e. in env sections, set VERSION={{.Version}}.)